corehq/motech/fhir/tests/test_fhir_views.py

```diff
@@ -183,6 +187,30 @@ def test_search(self):
         )
 
 
+class ViewsPermissionsTests(BaseFHIRViewTest):
+
+    def test_api_key_authenticated(self):
+        url = reverse(
+            "fhir_get_view",
+            args=[DOMAIN, FHIR_VERSION, "Patient", PERSON_CASE_ID],
+        )
+        request = _get_request(url, self.django_user, self.api_key.key)
+        is_authenticated = LoginAuthentication().is_authenticated(request)
```

Comment:
Nice! that's a smooth and small way of checking authentication. :+1: 

Just dropping thoughts here, and feel free to just punt this if it doesn't sound good enough.
An equivalent of these would be one test in the other test classes, `TestFHIRSearchView` and `TestFHIRGetView` with `request = _get_request(url, self.django_user)` with modification to `_get_request` method to get `api_key` as an argument instead. I say one because we it would just be the case when a user is not passing a valid api key. I don't think we should be adding a test for superuser, because its a feature that we have "removed" and does not look worth adding a test for.

Any thoughts on if that would be the right to do? I am thinking
1. the tests for now just check authentication on the `get` view and not the search view. If we remove the api check on the search view, tests would still pass.
2. we are kind of unit testing our hq authentication taking the get view as an instance. I believe the real intention here is to ensure that api is authenticated at the view properly.
3. We can also validate that we get back a 404 or whatever in the individual view tests, testing the complete behavior, which we aren't now.

Reply:
Thanks for breaking down your thoughts like this. I thought along similar lines ...

The existing `TestFHIRGetView` and `TestFHIRSearchView` classes already show the authentication "happy path" for both the "get" and "search" views. So the [Update tests](https://github.com/dimagi/commcare-hq/pull/29584/commits/538468de10dbc52bfd2b668ac52b0866eece56b6) commit takes advantage of that. To your point 3, I believe the existing tests cover the complete behavior already. (Am I right about that?) All that was missing was showing the difference in behavior before and after this code change.

So my intention with the new test class was just to verify my code change -- that the test fails before the change and passes afterwards.

You make a good point that the new test class only checks the "get" view, and should also verify the "search" view. I'll add that.

I'll take a look at `user_can_access_case`, and I'll also look at the behavior of the new Case API v.0.6, to confirm that our APIs behave consistently.

Reply:
> The existing TestFHIRGetView and TestFHIRSearchView classes already show the authentication "happy path" for both the "get" and "search" views. So the Update tests commit takes advantage of that. To your point 3, I believe the existing tests cover the complete behavior already. (Am I right about that?) All that was missing was showing the difference in behavior before and after this code change.

The existing tests are missing a test when user is not a superuser which should have checked for a 404. If it was there you probably could have modified those itself. 
I am just thinking that testing the authentication as a standalone test isn't the best way here. The views have been decorated with a method that returns a 404 if api key isn't provided, and is not aware of the internals for that decorator (that it does authentication) and we are testing that. So just saying that we should be testing what the view will return.

> So my intention with the new test class was just to verify my code change -- that the test fails before the change and passes afterwards.

Ya. I am just saying tests should be in the current classes itself and would check the behavior change.

Reply:
I've tried to clarify the scope of the test classes. (034e2be9954bf5d5ac64a57e2ca205a7dc1f6d39, fb115ee8200abe1825097e4dd80532395c6bc735 & 1a42b077c71c7471f702cdcfd30bed517074eebd)

I hope you'll agree that having a class specifically for authentication tests (which should have the same results for all views) reduces duplication.

I intend to build out the authorization tests a bit more, to show what cases a user can see and what they can't. If it turns out that it makes sense to move those tests into the get view tests, I might do that.